src/material-moment-adapter/adapter/index.ts

```diff
@@ -21,7 +25,15 @@ export * from './moment-date-formats';
 
 @NgModule({
   providers: [
-    {provide: DateAdapter, useClass: MomentDateAdapter, deps: [MAT_DATE_LOCALE]}
+    {
+      provide: DateAdapter,
+      useClass: MomentDateAdapter,
+      deps: [MAT_DATE_LOCALE, MAT_MOMENT_DATE_ADAPTER_OPTIONS]
+    },
+    {
+      provide: MAT_MOMENT_DATE_ADAPTER_OPTIONS,
```

Comment:
@jelbourn do you think we should use a root or module provider for this?

Reply:
Generally anything that doesn't depend on `Overlay` should be `providedIn: 'root'`

Reply:
You should be able to use `MAT_DATE_LOCALE` as an example: https://github.com/angular/material2/blob/master/src/lib/core/datetime/date-adapter.ts#L13

And tests for it here: https://github.com/angular/material2/blob/master/src/lib/core/datetime/native-date-adapter.spec.ts

It looks like you set it up correctly... what error were you getting in the tests?